src/trie/map.rs

```diff
@@ -187,8 +187,8 @@ impl<T> TrieMap<T> {
     /// assert_eq!(vec, vec!["c", "b"]);
     /// ```
     #[inline]
-    pub fn each_reverse<'a>(&'a self, f: |&uint, &'a T| -> bool) -> bool {
-        self.root.each_reverse(f)
+    pub fn each_reverse<F: FnMut(&uint, &T) -> bool>(&self, mut f: F) -> bool {
```

Comment:
I've been leaning towards a convention with Fn's all in Where clauses, personally.